pytorch_blade/src/ltc/disc_compiler/passes/cluster.cpp

```diff
@@ -11,14 +11,37 @@
 
 #include "ltc/disc_compiler/passes/cluster.h"
 
+#include <torch/csrc/jit/passes/dead_code_elimination.h>
 #include "compiler/jit/fusion.h"
 #include "compiler/mlir/converters/mhlo_conversion.h"
+#include "ltc/disc_compiler/passes/graph_fuser.h"
 
 #include <torch/script.h>
 namespace torch_disc {
 namespace compiler {
 using namespace ::torch::jit;
 
+bool IsDiscSupports(const torch::jit::Node* node) {
+  if (torch::blade::IsMlirMhloSupported(*node) &&
+      node->kind() != prim::Constant) {
+    for (auto& input : node->inputs()) {
+      // input should be Tensor or Scalar with explict type
+      auto typ = input->type();
+      if (!typ->cast<c10::TensorType>() &&
+          c10::tryScalarTypeFromJitType(*typ) == c10::nullopt) {
+        return false;
+      }
+    }
+    for (auto& output : node->outputs()) {
+      auto typ = output->type();
```

Comment:
By design, do operations that return non-Tensor type will not be supported? Such as, `split.Tensor(Tensor(a -> *) self, int split_size, int dim=0) -> Tensor(a)[]`.